### PR TITLE
Improve readability of `ifc.Loader`'s `Repo()` method

### PR DESCRIPTION
**Pull request by annasong20**
This PR aims to improve the readability of `ifc.Loader`'s `Repo()` method by simplifying its output. It addresses comment https://github.com/kubernetes-sigs/kustomize/pull/4797#discussion_r999693493 on PR #4797.

### CODE REVIEW

**Comment by annasong20** [comment]
`linter` currently failing due to previous change to master; #4854 should fix this, so I won't do anything here

File: api/internal/localizer/locloader.go
Comment (natasha41575) [inline api/internal/localizer/locloader.go:120]: nit: I think it will be more readable if we rename the above err to something like `fetchRemoteErr` and then checking if that is nil or not in the if conditions below, rather than creating a new bool here

File: api/internal/localizer/locloader.go
Comment (natasha41575) [inline api/internal/localizer/locloader.go:121]: The conditions here seem to be in the following order:

1. isRemote and repoSpec.Ref is empty
2. !isRemote
3. isRemote and repoSpec.Ref is not empty

This seems like kind of a weird order to put the code (separating the isRemote cases). I suggest nesting the if statements here, so it's more like

```
if isRemote {
  if repoSpec.Ref == "" {
  ...
  }
  else {
  ...
  }
} else {
  ... 
}
```

I think that would be easier to read the if branches.

Comment (natasha41575) [reply api/internal/localizer/locloader.go:121]: Oh, I looked a bit closer and see why it is structured the way it is. Maybe Yuwen's suggestion above to run the not remote code first would be a better way to structure it.

Comment (natasha41575) [reply api/internal/localizer/locloader.go:121]: Yuwen's suggestion above to do the not remote case first may also help with readability here.

**Comment by natasha41575** [comment]
My comments are just nits, so will defer to Yuwen to give her LGTM when the code seems readable enough. Feel free to remove the hold when you are ready.

/approve
/hold

File: api/internal/localizer/locloader.go
Comment (annasong20) [reply api/internal/localizer/locloader.go:119]: Lines 119-123 are a defensive check. 

The `ll.Loader.New(path)` on line 125 will try to `git fetch` the url. If there is no `ref`, this could take quite some time, potentially timing out the integration tests. As a result, I moved the check for `ref` before the call to `New()`.

Actually, though, there is a problem with the current implementation when `path` is a file url. I was going to introduce the change with a file url integration test in a separate PR. I can introduce that change here if you think it makes more sense. Please reference: https://github.com/kubernetes-sigs/kustomize/pull/4797/files#diff-3da63a4a59b5a528cb33383c63a905142e509f2c229d2fb4703c22bd008fb9f2R113-R120

File: api/internal/localizer/locloader.go
Comment (annasong20) [reply api/internal/localizer/locloader.go]: Hope it's easier to read now.

File: api/internal/localizer/util.go
Comment (yuwenma) [inline api/internal/localizer/util.go:97]: nit: `log.Fatalf("unable to parse validated root url: %s", err)`

File: api/internal/localizer/locloader.go
Comment (yuwenma) [inline api/internal/localizer/locloader.go:133]: nit: `'%s'` -> `%q`

**Review by yuwenma — APPROVED** [review]
Some super minor comments. Approved

**Pull request merged**
